opal/corelib/string.rb

```diff
@@ -1575,7 +1575,7 @@ def upto(stop, excl = false, &block)
     return enum_for :upto, stop, excl unless block_given?
     stop = Opal.coerce_to(stop, String, :to_str)
     %x{
-      var a, b, s = self.toString();
+      var a, b, s = self.toString(), value;
```

Comment:
@eregon I tried to look for how RubySpec currently handles testing methods that take a block which can have a `break` with a value called inside of it. I don't see much... e.g. https://github.com/ruby/rubyspec/blob/master/core/integer/times_spec.rb#L35 Am I missing something? Is this handled in some generic fashion elsewhere? Otherwise it looks like this case is very under-represented.